### ISSUE TITLE
Empty cell in flat-table results in exception

### ISSUE THREAD

**Issue by amykyta3** [return42/linuxdoc#3]
A recent commit seems to have broken flat-table's ability to describe an empty cell

The following:
```
FlatTable bug
=============

.. flat-table:: 
    :header-rows: 1
    
    *   - [1,1]
        - [2,1]
    
    *   - [1,2]
        - 
```

Results in:
```
Traceback (most recent call last):
  File "<hidden>/site-packages/sphinx/cmd/build.py", line 304, in build_main
    app.build(args.force_all, filenames)
  File "<hidden>/site-packages/sphinx/application.py", line 341, in build
    self.builder.build_update()
  File "<hidden>/site-packages/sphinx/builders/__init__.py", line 347, in build_update
    len(to_build))
  File "<hidden>/site-packages/sphinx/builders/__init__.py", line 360, in build
    updated_docnames = set(self.read())
  File "<hidden>/site-packages/sphinx/builders/__init__.py", line 468, in read
    self._read_serial(docnames)
  File "<hidden>/site-packages/sphinx/builders/__init__.py", line 490, in _read_serial
    self.read_doc(docname)
  File "<hidden>/site-packages/sphinx/builders/__init__.py", line 534, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "<hidden>/site-packages/sphinx/io.py", line 318, in read_doc
    pub.publish()
  File "<hidden>/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "<hidden>/site-packages/docutils/readers/__init__.py", line 72, in read
    self.parse()
  File "<hidden>/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "<hidden>/site-packages/sphinx/parsers.py", line 88, in parse
    self.statemachine.run(inputstring, document, inliner=self.inliner)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 171, in run
    input_source=document['source'])
  File "<hidden>/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "<hidden>/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 2753, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 395, in new_subsection
    node=section_node, match_titles=True)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 282, in nested_parse
    node=node, match_titles=match_titles)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "<hidden>/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "<hidden>/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 2753, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 395, in new_subsection
    node=section_node, match_titles=True)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 282, in nested_parse
    node=node, match_titles=match_titles)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "<hidden>/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "<hidden>/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 2326, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 2338, in explicit_construct
    return method(self, expmatch)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 2081, in directive
    directive_class, match, type_name, option_presets)
  File "<hidden>/site-packages/docutils/parsers/rst/states.py", line 2130, in run_directive
    result = directive_instance.run()
  File "<hidden>/site-packages/linuxdoc/rstFlatTable.py", line 123, in run
    tableBuilder.parseFlatTableNode(node)
  File "<hidden>/site-packages/linuxdoc/rstFlatTable.py", line 220, in parseFlatTableNode
    row = self.parseRowItem(rowItem, rowNum)
  File "<hidden>/site-packages/linuxdoc/rstFlatTable.py", line 339, in parseRowItem
    cspan, rspan, cellElements = self.parseCellItem(cellItem)
  File "<hidden>/site-packages/linuxdoc/rstFlatTable.py", line 351, in parseCellItem
    for elem in cellItem[0]:
  File "<hidden>/site-packages/docutils/nodes.py", line 569, in __getitem__
    return self.children[key]
IndexError: list index out of range

Exception occurred:
  File "<hidden>/site-packages/docutils/nodes.py", line 569, in __getitem__
    return self.children[key]
IndexError: list index out of range
```

**Comment by amykyta3** [comment]
Suspect it is related to this diff:
https://github.com/return42/linuxdoc/commit/d4ac0c353d2293f6e6a8a7fc77733de6fdd03efe#diff-54adc7a3ee4c624fd5bb5997e08209ac

**Comment by return42** [comment]
@amykyta3 thanks for your excellent issue report. I fixed this error with commit 7a3ef5e. Can you give it a try and report if it works for you?

**Comment by amykyta3** [comment]
Confirmed fixed.
Thanks for the quick turn-around!
